src/software/ai/navigator/obstacle/obstacle_factory_test.cpp

```diff
@@ -0,0 +1,173 @@
+#include "software/ai/navigator/obstacle/obstacle_factory.h"
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+
+#include "software/new_geom/circle.h"
+#include "software/new_geom/point.h"
+#include "software/new_geom/polygon.h"
+#include "software/new_geom/rectangle.h"
+#include "software/test_util/test_util.h"
+#include "software/world/robot.h"
+
+
+class ObstacleFactoryTest : public testing::Test
+{
+   public:
+    ObstacleFactoryTest()
+        : current_time(Timestamp::fromSeconds(123)),
+          obstacle_factory(std::make_shared<ObstacleFactoryConfig>())
```

Comment:
Won't using the `ObstacleFactoryConfig` use the default values generated from the YAML, so if we change those the values here will change? This would make the tests brittle.

Reply:
Specifically meant using the default constructor

Reply:
changed to using util dynamic param instead of default constructor. Out of scope to create specific dynamic params for these tests. I think it's blocked on the new dynamic params system, and then we'd want to change all the tests that use configs all at once